R/oauth.R

```diff
@@ -0,0 +1,727 @@
+#'   data must be included in either a JSON or an URL encoded request body:
+#'   - `grant_type`, this must be `authorization_code` for new tokens,
+#'     and `refresh_token` for refreshing.
+#'   - `code`, this must be the temporary code obtained from the
+#'     `/authorize/decision` redirection, for new tokens. It is not
+#'     needed when refreshing.
+#'   - `client_id` must be the client id of the third party app.
+#'   - `client_secret` must be the client secret of the third party
+#'     app.
+#'   - `redirect_uri` must be the correct redirection URI of the
+#'     third party app. It is not needed when refreshing tokens.
+#'   - `refresh_token` must be the refresh token obtained previously,
+#'     when refreshing a token. It is not needed for new tokens.
+#'   On success a JSON dictionary is returned with entries:
+#'   `access_token`, `expiry` and `refresh_token`. (The latter is
+#'   omitted if the `refresh` argument is `FALSE`).
+#' * `GET /locals` returns a list of current apps, access tokens and
+#'   refresh tokens.
+#' * `GET /data` is an endpoint that returns a simple JSON response,
+#'   and needs authorization.
+#'
+#' ## Notes
+#'
+#' * Using this app in your tests requires the glue package, so you
+#'   need to put it in `Suggests`.
+#' * You can add custom endpoints to the app, as needed.
+#' * If you need authorization in your custom endpoint, call
+#'   `app$is_authorized()` in your handler:
+#'   ```
+#'   if (!app$is_authorized(req, res)) return()
+#'   ```
+#'   `app$is_authorized()` returns an HTTP 401 response if the
+#'   client is not authorized, so you can simply return from your
+#'   handler.
+#'
+#' For more details see `vignette("oauth", package = "webfakes")`.
+#'
+#' @section `oauth2_resource_app()`:
+#' App representing the API server (resource/authorization)
+#' @return a `webfakes` app
+#' @param access_duration After how many seconds should access tokens
+#'   expire.
+#' @param refresh_duration After how many seconds should refresh
+#'   tokens expire (ignored if `refresh` is `FALSE`).
+#' @param refresh Should a refresh token be returned (logical).
+#' @param seed Random seed used when creating tokens. If `NULL`,
+#'   we rely on R to provide a seed. The app uses its own RNG stream,
+#'   so it does not affect reproducibility of the tests.
+#' @param authorize_ep The authorization endpoint of the resource
+#'   server. Change this from the default if the real app that you
+#'   are faking does not use `/authorize`.
+#' @param token_ep The endpoint to request tokens. Change this if the
+#'   real app that you are faking does not use `/token`.
+#' @return webfakes app
+#'
+#' @export
+#' @family OAuth2.0 functions
+
+oauth2_resource_app <- function(access_duration = 3600L,
+                                refresh_duration = 7200L,
+                                refresh = TRUE, seed = NULL,
+                                authorize_ep = "/authorize",
```

Comment:
I know `authorize_ep` is shorter but I'd find it easier if it were called `authorize_endpoint`

Reply:
I'll rename that and the other shortened argument